.github/testflinger/job-def.yaml

```diff
@@ -0,0 +1,116 @@
+job_queue: REPLACE_QUEUE
+output_timeout: 7200
+provision_data:
+  REPLACE_PROVISION_DATA
+test_data:
+  test_cmds: |
+
+    # Exit immediately if a test fails
+    set -e
+
+    # Clone repo from appropriate branch/commit.
+    ssh ubuntu@$DEVICE_IP '
+      sudo DEBIAN_FRONTEND=noninteractive apt update
+      sudo DEBIAN_FRONTEND=noninteractive apt -y upgrade
+      sudo DEBIAN_FRONTEND=noninteractive apt -y install git curl
+      git clone -b REPLACE_BRANCH \
+        https://github.com/canonical/intel-npu-driver-snap.git \
+        ~ubuntu/intel-npu-driver-snap
+      cd ~ubuntu/intel-npu-driver-snap
+      echo "Current git branch: $(git branch --show-current)"
+      echo "Latest commit:"
+      git log --name-status HEAD^..HEAD
+    '
+
+    # Install dependencies
+    ssh ubuntu@$DEVICE_IP '
+      sudo snap install --classic snapcraft
+      sudo snap install lxd --channel=5.21/stable
+      sudo adduser ubuntu lxd
+      sudo snap refresh
+    '
+
+    # Disable swap and IPv6
+    ssh ubuntu@$DEVICE_IP '
+      sudo sysctl -w vm.swappiness=0
+      sudo echo "vm.swappiness = 0" | sudo tee -a /etc/sysctl.conf
+      sudo swapoff -a
+      echo "net.ipv6.conf.all.disable_ipv6=1" | sudo tee -a /etc/sysctl.conf
+      echo "net.ipv6.conf.default.disable_ipv6=1" | sudo tee -a /etc/sysctl.conf
+      echo "net.ipv6.conf.lo.disable_ipv6=1" | sudo tee -a /etc/sysctl.conf
+      sudo sysctl -p
+    '
+
+    # Build and install intel-npu-driver snap
+    ssh ubuntu@$DEVICE_IP '
+      lxd init --auto
+      cd ~ubuntu/intel-npu-driver-snap
+      snapcraft
+      sudo snap install --dangerous ./intel-npu-driver_*_amd64.snap
+      sudo snap connect intel-npu-driver:intel-npu-fw
+      sudo snap connect intel-npu-driver:intel-npu-kmod
+      sudo snap connect intel-npu-driver:intel-npu-plug intel-npu-driver:intel-npu
+    '
+
+    echo "[INFO]: Sleeping for 60 seconds to ensure the firmware search path is updated."
+    sleep 60
```

Comment:
this means that the hook is executed asynchronously ? Does it mean that if user execute too quickly the test app or other consumer app, they can expect failure or even worse, they have a success but they might be using the old firmware

Reply:
Recall that the firmware is now installed and loaded [with this daemon](https://github.com/canonical/intel-npu-driver-snap/blob/main/snap/snapcraft.yaml#L58-L66). The daemon runs every 10 seconds. What you describe is technically possible, and I'd like to improve the design, but it is challenging to manage the firmware from a hook (see previous discussion from [this PR](https://github.com/canonical/intel-npu-driver-snap/pull/8)).

I think in practice because the snap is a producer snap that it will be installed prior to a consumer snap, making it unlikely that an application would run before the firmware is ready, but I do agree that it is not an ideal design.

About the actual test, 60 seconds is somewhat arbitrary conservative value.

Reply:
Ah i recall it now, sorry, the hook does not work on reboot,